Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -1795,18 +1798,23 @@ - (void)_webTouchEventsRecognized:(UIWebTouchEventsGestureRecognizer *)gestureRe
         if (nativeWebTouchEvent.isPotentialTap() && self.hasHiddenContentEditable && self._hasFocusedElement && !self.window.keyWindow)
             [self.window makeKeyWindow];
 
-        auto stopDeferringNativeGesturesIfNeeded = [] (NSArray<WKDeferringGestureRecognizer *> *deferringGestures) {
-            for (WKDeferringGestureRecognizer *gesture in deferringGestures) {
-                if (gesture.state == UIGestureRecognizerStatePossible)
-                    gesture.state = UIGestureRecognizerStateFailed;
-            }
+        auto stopDeferringNativeGesturesIfNeeded = [] (WKDeferringGestureRecognizer *gestureRecognizer) {
+            if (gestureRecognizer.state == UIGestureRecognizerStatePossible)
+                gestureRecognizer.state = UIGestureRecognizerStateFailed;
         };
 
-        if (!_page->isHandlingPreventableTouchStart())
-            stopDeferringNativeGesturesIfNeeded(self._touchStartDeferringGestures);
+        if (!_page->isHandlingPreventableTouchStart()) {
+            for (WKDeferringGestureRecognizer *gestureRecognizer in self._touchStartDeferringGestures)
+                stopDeferringNativeGesturesIfNeeded(gestureRecognizer);
+        }
 
-        if (!_page->isHandlingPreventableTouchEnd())
-            stopDeferringNativeGesturesIfNeeded(self._touchEndDeferringGestures);
+        if (!_page->isHandlingPreventableTouchEnd()) {
+            for (WKDeferringGestureRecognizer *gestureRecognizer in self._touchEndDeferringGestures)
+                stopDeferringNativeGesturesIfNeeded(gestureRecognizer);
+        }
+
+        if (!_page->isHandlingPreventableTouchMove())
+            stopDeferringNativeGesturesIfNeeded(_touchMoveDeferringGestureRecognizer.get());
```

Comment:
NIT: any reason this cant be moved in between the stuff for `"touchstart"` and the stuff for `"touchend"`?

Reply:
No reason one way or another — it should not make a difference.